services/core-web/src/components/common/RenderAutoSizeField.js

```diff
@@ -16,37 +16,60 @@ const propTypes = {
   meta: PropTypes.objectOf(PropTypes.any).isRequired,
   disabled: PropTypes.bool,
   minRows: PropTypes.number,
+  maximumCharacters: PropTypes.number,
 };
 
 const defaultProps = {
   placeholder: "",
   label: "",
   disabled: false,
   minRows: 3,
+  maximumCharacters: 300,
 };
 
-const RenderAutoSizeField = (props) => (
-  <Form.Item
-    label={props.label}
-    placeholder={props.placeholder}
-    validateStatus={
-      props.meta.touched ? (props.meta.error && "error") || (props.meta.warning && "warning") : ""
-    }
-    help={
-      props.meta.touched &&
-      ((props.meta.error && <span>{props.meta.error}</span>) ||
-        (props.meta.warning && <span>{props.meta.warning}</span>))
-    }
-  >
-    <Input.TextArea
-      disabled={props.disabled}
-      id={props.id}
-      {...props.input}
-      autoSize={{ minRows: props.minRows }}
+const RenderAutoSizeField = (props) => {
+  const [remainingChars, setRemainingChars] = useState(props.maximumCharacters);
+  const [value, setValue] = useState("");
+
+  const handleTextAreaChange = (event) => {
+    setValue(event.target.value);
+
+    const input = event.target.value;
+    const remaining = props.maximumCharacters - input.length;
+    setRemainingChars(remaining);
+  };
+
+  return (
+    <Form.Item
+      label={props.label}
       placeholder={props.placeholder}
-    />
-  </Form.Item>
-);
+      validateStatus={
+        props.meta.touched ? (props.meta.error && "error") || (props.meta.warning && "warning") : ""
+      }
+      help={
+        props.meta.touched &&
+        ((props.meta.error && <span>{props.meta.error}</span>) ||
+          (props.meta.warning && <span>{props.meta.warning}</span>))
+      }
+    >
+      <Input.TextArea
+        disabled={props.disabled}
+        id={props.id}
+        {...props.input}
+        autoSize={{ minRows: props.minRows }}
+        placeholder={props.placeholder}
+        onChange={handleTextAreaChange}
+        value={value}
```

Comment:
Cool, this should definitely work.  I know AntD does have a [`showCount`](https://ant.design/components/input#components-input-demo-show-count) prop for the `Input.TextArea` that might make it a little bit simpler.  

I haven't played with it myself though so maybe that doesn't provide quite the behaviour we're looking for.  Did you look into that at all?